webpack/assets/javascripts/react_app/common/MountingService.js

```diff
@@ -47,25 +54,40 @@ class ReactComponentElement extends HTMLElement {
     return this._mountPoint;
   }
 
+  attributeChangedCallback(name, oldValue, newValue) {
+    switch (name) {
+      case 'data-props':
+        // if this is not the initial prop set
+        if (oldValue) this._render();
```

Comment:
what if the previous value was `{}` or some other false-y value?

Reply:
also, do we need to unmount the old component before re-mounting or is this done automatically when we mount a new component at the same place?

Reply:
No, we do not need to unmount, the component gets refreshed correctly, by the same call. But you're right that we should be more specific with the oldValue.

Reply:
Comparing with null now as that is the default value before js reads the dom attribute.